File: src/main/java/com/example/CargoTracking/model/Location.java
package com.example.CargoTracking.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.*;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.List;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@ToString
@Builder
@Entity
public class Location {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String locationName;
    private String type;
    private boolean status;
    private String originEmail;
    private String destinationEmail;
    private String originEscalationLevel1;
    private String originEscalationLevel2;
    private String originEscalationLevel3;
    private String destinationEscalationLevel1;
    private String destinationEscalationLevel2;
    private String destinationEscalationLevel3;

    @ManyToOne
    @JoinColumn(name = "country_id")
    private Country country;

    @ManyToOne
    @JoinColumn(name = "facility_id")
    private Facility facility;

    @Override
    public String toString() {
        return "Location{" +
                "id=" + id +
                ", locationName='" + locationName + '\'' +
                ", type='" + type + '\'' +
                ", status=" + status +
                ", originEmail='" + originEmail + '\'' +
                ", destinationEmail='" + destinationEmail + '\'' +
                '}';
    }
}

File: frontend/src/app/model/Location.ts
import { Country } from "./Country";
import { DestinationEmails, EscalationEmails, OriginEmails } from "./Email";
import { Facility } from "./Facility";


export interface Location{

    id: number|undefined|null;
    locationName: string|undefined|null;
    type: string|undefined|null;
    originEmail:string[]|string|null;
    destinationEmail:string[]|string|null;
    originEscalationLevel1:string[]|string|null;
    originEscalationLevel2:string[]|string|null;
    originEscalationLevel3:string[]|string|null;
    destinationEscalationLevel1:string[]|string|null;
    destinationEscalationLevel2:string[]|string|null;
    destinationEscalationLevel3:string[]|string|null;
    status: boolean|undefined|null;
    facility: Facility|undefined|null;
    country: Country|undefined|null;
   
}